### PR TITLE
Add popup showing Today's Appointments

### PR DESCRIPTION
**Pull request by Dominic-Khoo**
Added Popup window that shows a list of patients with appointments on the current date. Pop-up will NOT appear if there are no patients with appointments on the current date.

### CODE REVIEW

File: src/main/java/seedu/address/ui/AppointmentPopup.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    /**
+     * Checks if there are any appointments scheduled for today among the filtered list of persons.
+     * If appointments are found, they are displayed in the popup.
+     *
+     * @return {@code true} if there are appointments today, otherwise {@code false}.
+     */
+    public boolean hasAppointmentsToday() {
+        LocalDate currentDate = LocalDate.now();
+        List<Person> personsWithAppointments = filteredPersonList.stream()
```
Comment (reidenong) [inline src/main/java/seedu/address/ui/AppointmentPopup.java:60]: functional code is clean and easy to read, nice

File: src/main/java/seedu/address/model/person/Appointment.java [hunk 2]
```diff
@@ -8,15 +9,21 @@
  */
 public class Appointment {
 
+    /** Message constraints that indicate the valid format for appointments. */
```
Comment (reidenong) [inline src/main/java/seedu/address/model/person/Appointment.java:12]: Nice improvements to the documentation of the appointment class

**Review by reidenong — APPROVED** [review]
Overall LGTM!

File: src/main/java/seedu/address/model/person/Appointment.java [hunk 3]
```diff
@@ -8,15 +9,21 @@
...
+
+    /** Formatter for displaying and parsing appointment date and time. */
     public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
 
+    /** The date and time value of the appointment. */
     public final LocalDateTime value;
 
     /**
-     * Constructs an {@code Appointment}.
+     * Constructs an {@code Appointment} with the specified date and time string.
```
Comment (itsnotsherm) [inline src/main/java/seedu/address/model/person/Appointment.java:26]: Added clarity is always nice

File: src/main/java/seedu/address/ui/AppointmentPopup.java
Comment (itsnotsherm) [inline src/main/java/seedu/address/ui/AppointmentPopup.java:1]: clean code

**Review by itsnotsherm — APPROVED** [review]
LGTM, just need update test cases to cover added LOC in Appointment class

**Pull request merged**